### PR TITLE
Improve Coder encode/decode documentation.

### PR DESCRIPTION
**Pull request by arunpandianp**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by arunpandianp** [comment]
R: @reuvenlax

**Comment by arunpandianp** [comment]
Run Spotless PreCommit

**Comment by arunpandianp** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by damccorm** [comment]
@johnjcasey or @Abacn would one of you mind taking a look?

@arunpandianp it looks like you have formatting errors that can be fixed by running `./gradlew :sdks:java:core:spotlessApply` locally

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/Coder.java [hunk 1]
```diff
@@ -113,6 +113,9 @@ public String toString() {
 
   /**
    * Encodes the given value of type {@code T} onto the given output stream.
+   * Multiple elements can be encoded next to each other on the output stream,
+   * each coder should encode information to know how many bytes to read when decoding.
+   * A common approach is to prefix the encoding with the elemen's encoded length.
```
Comment (johnjcasey) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/Coder.java:118]: typo in element on line 118

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/Coder.java [hunk 2]
```diff
@@ -135,6 +138,9 @@ public void encode(T value, OutputStream outStream, Context context)
   /**
    * Decodes a value of type {@code T} from the given input stream in the given context. Returns the
    * decoded value.
+   * Multiple elements can be encoded next to each other on the input stream,
+   * each coder should encode information to know how many bytes to read when decoding.
+   * A common approach is to prefix the encoding with the elemen's encoded length.
```
Comment (johnjcasey) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/Coder.java:143]: typo is duplicated here

**Comment by arunpandianp** [comment]
Fixed the typos and ran spotless. Pre commit Java is failing, not sure why. Shouldn't be due to this comment only change.

**Pull request merged**